syntax = "proto3";
package logtail.models;

message SpanEvent {
    uint64 Timestamp = 1;
    string TraceID = 2;
    string SpanID = 3;
    string TraceState = 4;
    string ParentSpanID = 5;
    string Name = 6;
    enum SpanKind {
        UNSPECIFIED = 0;
        INTERVAL = 1;
        SERVER = 2;
        CLIENT = 3;
        PRODUCER = 4;
        CONSUMER = 5;
    }
    SpanKind Kind = 7;
    uint64 StartTimeNs = 8;
    uint64 EndTimeNs = 9;
    map<string, string> Tags = 10;
    message InnerEvent {
        uint64 TimestampNs = 1;
        string Name = 2;
        map<string, string> Tags = 3;
    }
    repeated InnerEvent Events = 11;
    message SpanLink {
        string TraceID = 1;
        string SpanID = 2;
        string TraceState = 3;
        map<string, string> Tags = 4;
    }
    repeated SpanLink Links = 12;
    enum StatusCode {
        Unset = 0;
        Ok = 1;
        Error = 2;
    }
    StatusCode Status = 13;
    map<string, string> ScopeTags = 14;
}